File: php/src/Invoice.php
<?php

class Invoice
{
    public $amount;
    public $currency;
    public string $customer;

    public function __construct(int $amount, \Currency $currency, string $customer)
    {
        $this->amount = $amount;
        $this->currency = $currency;
        $this->customer = $customer;
    }
}

File: csharp/lib/Invoice.cs
namespace lib
{
    public class Invoice
    {
        public int Amount { get; }
        public string Currency { get; }
        public string CompanyName { get; }

        public Invoice(int amount, string currency, string companyName)
        {
            Amount = amount;
            Currency = currency;
            CompanyName = companyName;
        }
    }
}